src/BuildingRegistry/Building/Building.cs

```diff
@@ -236,6 +236,26 @@ public void ChangeOutliningConstruction(ExtendedWkbGeometry extendedWkbGeometry)
                 buildingUnitsPosition));
         }
 
+        public void MarkAsMerged(BuildingPersistentLocalId destinationBuildingPersistentLocalId)
+        {
+            foreach (var buildingUnit in _buildingUnits.PlannedBuildingUnits()
+                         .Concat(_buildingUnits.RealizedBuildingUnits())
+                         .Where(x => x.Function != BuildingUnitFunction.Common)
+                         .ToList())
+            {
+                ApplyChange(new BuildingUnitWasMoved(
+                    BuildingPersistentLocalId,
+                    buildingUnit.BuildingUnitPersistentLocalId,
+                    destinationBuildingPersistentLocalId));
+
+                NotRealizeOrRetireCommonBuildingUnit();
```

Comment:
Can we move this after the loop? We know its going to be NotRealized or Retired. If we move it after the loop, we don't have to check with each iteration.

Reply:
No, we need to respect the state of the building. From the moment one `buildingunit` remains, the common needs to disappear.

Reply:
True.